packages/styles/section-header.css

```diff
@@ -0,0 +1,54 @@
+:root {
+  --section-header-description-color: var(--gray-60);
+  --section-header-heading-color: var(--gray-90);
+}
+
+.cauldron--theme-dark {
+  --section-header-description-color: var(--accent-light);
+  --section-header-heading-color: #ffffff;
+}
+
+.SectionHeader {
+  container-type: inline-size;
+}
+
+.SectionHeader__content {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  padding-bottom: var(--space-large);
+  gap: var(--space-small);
+}
+
+.SectionHeader__text-section {
+  display: flex;
+  flex-direction: column;
+  gap: var(--space-smallest);
+  flex: 1;
+}
```

Comment:
I _think_ we can eliminate this class/wrapper. We should just be able to have:

```tsx
<div className="SectionHeader__content">
  {heading}
  {description && (
    <p className="SectionHeader__description">{description}</p>
  )}
  {children && <div className="SectionHeader__actions">{children}</div>}
</div>
```

We should be able to use grid instead of flex to align the items without the extra wrapper.